### PR TITLE
add window capture support

### LINKED ISSUE #41: Implement window capture
On `x11grab` the `window_id` input option is available, while on `gdigrab` the window title can be provided, but i need to look further into using a HWND instead of a title

### PR DESCRIPTION
**Pull request by adryzz**
Closes #41 

- [ ] `x11grab` window capture
- [x] `gdigrab` window capture

### CODE REVIEW

File: SeeShark/VideoInputOptions.cs [hunk 1]
```diff
@@ -59,6 +59,11 @@ public class VideoInputOptions
     /// </summary>
     public bool DrawMouse { get; set; } = true;
 
+    /// <summary>
+    /// Used in Linux only - The ID of the window to capture
+    /// </summary>
+    public string? WindowId { get; set; }
```
Comment (sr229) [inline SeeShark/VideoInputOptions.cs:65]: Probably make it throw `PlatformNotSupportedException` if someone bothers to call this on a non-Linux system.

Comment (adryzz) [reply SeeShark/VideoInputOptions.cs:65]: it is set on both windows and linux (On windows it's the HWND of the window while on linux it's the window id, but ffmpeg has no way of capturing a window on windows (pun unintended) using the HWND, but it uses the window title, which means that it simply gets ignored

Comment (sr229) [reply SeeShark/VideoInputOptions.cs:65]: Being explicit is better than implicit here, so I think we still need to inform the programmer they don't need to do this.

**Comment by adryzz** [comment]
`gdigrab` works, although i've encountered a weird issue on the decoder thread where the calculation for how much time to sleep divides by 0 (e.g. `decoder.Framerate.den` is equal to `0`)

image

**Comment by adryzz** [comment]
Apparently, the decoder frame rate is just a guess from FFmpeg, so when it's 0 we cam just fall back to a default value, as the worst that's gonna happen is a little higher CPU usage

**Comment by adryzz** [comment]
Apparently `gdigrab` cannot grab UWP windows...

https://trac.ffmpeg.org/ticket/7718

**Comment by adryzz** [comment]
i did not forget about this PR, finally gotten around to submitting the [ffmpeg patch](http://ffmpeg.org/pipermail/ffmpeg-devel/2023-December/318122.html) this depends on, we'll see how it goes from there

**Comment by adryzz** [comment]
WE ARE IN LETS GOOOOOOOOOOO

https://github.com/FFmpeg/FFmpeg/commit/d7ac7101ee8c70a253f7caf400d9ae3870880d7f

ill finally finish this pr lol

**Comment by adryzz** [comment]
welp it's definitely in mainline by now, time to update and stuff

**Comment by Speykious** [comment]
Seems like I got the time to make a complete rewrite PR before this one got updated. xD
I think it's best I close this one in favor of #51. Though I'm sure we can reuse all the interop code for GDI support.

**Pull request closed without merging**